Plugins/NativeCamera/Source/NativeCamera.cpp

```diff
@@ -0,0 +1,166 @@
+                global.Set(JS_NAVIGATOR_NAME, navigator);
+            }
+
+            // append media devices to navigator
+            Napi::Object mediaDevices = Napi::Object::New(env);
+            mediaDevices.Set("getUserMedia", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto env = info.Env();
+                auto deferred{ Napi::Promise::Deferred::New(env) };
+                auto promise{ deferred.Promise() };
+               
+                auto& jsRuntime{ JsRuntime::GetFromJavaScript(env) };
+                    jsRuntime.Dispatch([deferred{ std::move(deferred) }](Napi::Env env) {
+                        deferred.Resolve(env.Null());
+                    });
+                
+                return promise;
+                }));
+            navigator.Set("mediaDevices", mediaDevices);
+        }
+
+        NativeCamera(const Napi::CallbackInfo& info)
+            : Napi::ObjectWrap<NativeCamera>{ info }
+        {
+        }
+
+        ~NativeCamera()
+        {
+        }
+
+    private:
+        Napi::Value CreateVideo(const Napi::CallbackInfo& info)
+        {
+            const auto env = info.Env();
+            Napi::Object video = Napi::Object::New(env);
+            
+            auto constraints = info[0].As<Napi::Object>();
+            uint32_t maxWidth{256}, maxHeight{256};
+            std::string facingMode;
+
+            auto maxWidthValue = constraints.Get("maxWidth");
+            auto maxHeightValue = constraints.Get("maxHeight");
+            auto facingModeValue = constraints.Get("facingMode");
+            if (maxWidthValue.IsNumber())
+            {
+                maxWidth = maxWidthValue.As<Napi::Number>().Uint32Value();
+            }
+            if (maxHeightValue.IsNumber())
+            {
+                maxHeight = maxHeightValue.As<Napi::Number>().Uint32Value();
+            }
+            if (facingModeValue.IsString())
+            {
+                facingMode = facingModeValue.As<Napi::String>().Utf8Value();
+            }
+
+            video.Set("videoWidth", Napi::Number::New(env, maxWidth).As<Napi::Value>());
+            video.Set("videoHeight", Napi::Number::New(env, maxHeight).As<Napi::Value>());
+            video.Set("frontCamera", Napi::Boolean::New(env, (facingMode == "user")).As<Napi::Value>());
+            video.Set("readyState", Napi::Number::New(env, 10).As<Napi::Value>());
+            video.Set("HAVE_CURRENT_DATA", Napi::Number::New(env, 1).As<Napi::Value>());
+            video.Set("isNative", Napi::Boolean::New(env, true).As<Napi::Value>());
+
+            video.Set("setAttribute", Napi::Function::New(env,[](const Napi::CallbackInfo& /*info*/){
+                }));
+
+            video.Set("addEventListener", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto callback = info[1].As<Napi::Function>();
+                info.This().ToObject().Set("_cb", callback.As<Napi::Object>().As<Napi::Value>());
```

Comment:
I'm not sure I understand this pattern. It looks like it's only possible for there to be a single callback here regardless of the argument at `info[0]`. Is this the intent?

Reply:
How are there two of this comment?